crates/emitter/src/forward_jump_emitter.rs

```diff
@@ -0,0 +1,33 @@
+use super::emitter::{BytecodeOffset, EmitError};
+use crate::ast_emitter::AstEmitter;
+
+// Struct for emitting bytecode for forward jump.
+#[derive(Clone, Copy, Debug)]
+pub struct ForwardJumpEmitter<F>
+where
+    F: Fn(&mut AstEmitter) -> Result<(), EmitError>,
+{
+    pub offset: BytecodeOffset,
+    pub jump: F,
+}
+impl<F> ForwardJumpEmitter<F>
+where
+    F: Fn(&mut AstEmitter) -> Result<(), EmitError>,
+{
+    #[must_use]
+    pub fn emit(self, emitter: &mut AstEmitter) -> Result<(), EmitError> {
+        (self.jump)(emitter)?;
+
+        // The JITs rely on a jump target being emitted after the
+        // conditional jump
+        emitter.emit.jump_target();
+        return Ok(());
```

Comment:
remove `return` and trailing semi-colon.
the last expression value becomes the return value.